Compiler warning while building HPC-GAP under Cygwin

The following is observed on Jenkins cygwin and cygwin64 nodes, with gcc 4.9.2 and 4.9.3 respectively:
```
src/system.c: In function â€˜storePosIntegerâ€™:
src/system.c:979:3: error: array subscript has type â€˜charâ€™ [-Werror=char-subscripts]
  while (isdigit(*p)) {
  ^
src/system.c: At top level:
cc1: error: unrecognized command line option "-Wno-implicit-fallthrough" [-Werror]
cc1: all warnings being treated as errors
Makefile.rules:347: recipe for target 'obj/src/system.lo' failed
make: *** [obj/src/system.lo] Error 1
Build step 'Execute shell' marked build as failure
```

--- a/src/system.c
+++ b/src/system.c
@@ -976,7 +976,7 @@ static Int storePosInteger( Char **argv, void *Where )
   UInt n;
   Char *p = argv[0];
   n = 0;
-  while (isdigit(*p)) {
+  while (IsDigit(*p)) {
     n = n * 10 + (*p-'0');
     p++;
   }

